docker/jupyter_notebook_config.py

```diff
@@ -0,0 +1,17 @@
+# pulled from https://hub.docker.com/r/ansingh7115/leonardo-notebooks/
+# and added base_url configs for local proxy testing
+
+from jupyter_core.paths import jupyter_data_dir
+import subprocess
+import os
+import errno
+import stat
+
+c = get_config()
+c.NotebookApp.ip = '*'
+c.NotebookApp.port = 8000
+c.NotebookApp.open_browser = False
+c.NotebookApp.base_url = '/notebooks/test/'
```

Comment:
So we want our proxy to live at the path:
```
http://leo:8080/notebooks/{id}
```
It gets immensely easier if the Notebook server is listening on the same path. Otherwise the client-side notebook code tries to load assets at the root path (http://leo:8080/) which does not get proxied.

I admit I don't fully understand all the calls being made, but I had problems with other configurations. Things I could try:
1. Look into exactly what `NotebookApp.base_url` is doing in Jupyter code. Is there some way we can simulate this in our reverse proxy, and avoid changing `jupyter_notebook_config.py`?
2. Make the Leo proxy listen at / as well as /notebooks/id. The root path might just be needed for static resources. Need to make sure it doesn't clash with Swagger which also listens at /.